src/WebJobs.Script/Host/ScriptTelemetryClientFactory.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Reflection;
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.Extensibility.Implementation;
+using Microsoft.ApplicationInsights.WindowsServer.Channel.Implementation;
+using Microsoft.Azure.WebJobs.Host.Loggers;
+
+namespace Microsoft.Azure.WebJobs.Script.Host
+{
+    /// <summary>
+    /// Overrides the default client creation by adding a custom SdkVersion for backend tracking.
+    /// </summary>
+    internal class ScriptTelemetryClientFactory : ITelemetryClientFactory
+    {
+        private const string Unknown = "[Unknown]";
+
+        public TelemetryClient Create(string instrumentationKey, SamplingPercentageEstimatorSettings samplingSettings)
+        {
+            ITelemetryClientFactory defaultFactory = new DefaultTelemetryClientFactory();
+            TelemetryClient client = defaultFactory.Create(instrumentationKey, samplingSettings);
+
+            string assemblyVersion = GetAssemblyFileVersion(typeof(ScriptHost).Assembly);
+            client.Context.GetInternalContext().SdkVersion = $"azurefuncs: {assemblyVersion}";
+
+            return client;
+        }
+
+        internal static string GetAssemblyFileVersion(Assembly assembly)
```

Comment:
Don't we already have logic that gets the appropriate runtime version somewhere (ends up in the Version property of the host)? Can we reuse that instead?